experimental/examples/imageviewer/shared/src/commonMain/kotlin/example/imageviewer/view/MainScreen.kt

```diff
@@ -1,71 +1,211 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
 import example.imageviewer.Dependencies
-import example.imageviewer.model.*
-import example.imageviewer.model.State
-import example.imageviewer.style.*
+import example.imageviewer.model.GalleryScreenState
+import example.imageviewer.model.Picture
+import example.imageviewer.model.PictureWithThumbnail
+import example.imageviewer.model.bigUrl
+import example.imageviewer.style.ImageviewerColors
+import example.imageviewer.style.ImageviewerColors.kotlinHorizontalGradientBrush
 import org.jetbrains.compose.resources.ExperimentalResourceApi
 import org.jetbrains.compose.resources.orEmpty
 import org.jetbrains.compose.resources.rememberImageBitmap
 import org.jetbrains.compose.resources.resource
 
 @Composable
-internal fun MainScreen(state: MutableState<State>, dependencies: Dependencies) {
-    Column {
-        TopContent(state, dependencies)
-        ScrollableColumn(
-            modifier = Modifier.fillMaxSize()
-        ) {
-            for (i in state.value.pictures.indices) {
-                val picture = state.value.pictures[i]
-                Miniature(
-                    picture = picture,
-                    image = state.value.miniatures[picture],
-                    onClickSelect = {
-                        state.setSelectedIndex(i)
-                    },
-                    onClickFullScreen = {
-                        state.toFullscreen(i)
-                    },
-                    onClickInfo = {
-                        dependencies.notification.notifyImageData(picture)
-                    },
-                )
-                Spacer(modifier = Modifier.height(4.dp))
+internal fun GalleryHeader() {
+    Row(
+        horizontalArrangement = Arrangement.Center,
+        modifier = Modifier.padding(10.dp).fillMaxWidth()
+    ) {
+        Text(
+            "My Gallery",
+            fontSize = 25.sp,
+            color = MaterialTheme.colorScheme.onBackground,
+            fontStyle = FontStyle.Italic
+        )
+    }
+}
+
+enum class GalleryStyle {
+    SQUARES,
+    LIST
+}
+
+fun GalleryStyle.toggled(): GalleryStyle {
+    return if (this == GalleryStyle.SQUARES) GalleryStyle.LIST else GalleryStyle.SQUARES
+}
+
+@Composable
+internal fun MainScreen(galleryScreenState: GalleryScreenState, dependencies: Dependencies) {
+    var galleryStyle by remember { mutableStateOf(GalleryStyle.SQUARES) } // TODO: Potentially expose a toggle for this.
```

Comment:
As now there is no switch for LIST view, I suspect it will never appear. So I would  suggest either to add the SQARES/LIST switch on the bottom of the screen or remove List view.

Reply:
Toggle added in top bar! :)
![2023-02-15 15 31 09](https://user-images.githubusercontent.com/2178959/219056543-ffd2734d-809e-4280-b08c-3b7b95dfb3b2.gif)